airship_tempest_plugin/tests/api/shipyard/test_actions.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2018 AT&T Corp
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+import logging
+
+from airship_tempest_plugin.tests.api.shipyard import base
+
+from tempest import config
+from tempest.lib import decorators
+
+CONF = config.CONF
+LOG = logging.getLogger(__name__)
+
+
+class ActionsTest(base.BaseShipyardTest):
+
+    def _get_action_id(self):
+        resp = self.shipyard_actions_client.list_actions()
+        self.assertTrue(len(resp[1]) > 0,
+                        'No actions available, nothing to test')
+        # get the response body
+        return resp[1]['id']
+
+    def _get_action_step_id(self):
+        resp = self.shipyard_actions_client.list_actions()
+        self.assertTrue(len(resp[1]) > 0,
+                        'No actions available, nothing to test')
+        return resp[1]['id'], resp[1]['steps'][0]['id']
```

Comment:
I don't think list_actions() will return any information about steps. I think to get step information, you need to describe/get a particular action:

`./run_shipyard.sh get actions
Name               Action                                   Lifecycle        Execution Time             Step Succ/Fail/Oth        
deploy_site        action/01CR787N8G784NDTWY6C20MGTD        Complete         2018-09-25T02:16:54        12/2/3                    
deploy_site        action/01CR8JPKBFF1CWADJ0B9TATSXB        Complete         2018-09-25T14:39:05        12/2/3                    
deploy_site        action/01CR8QEZAVCN22HZN0EN6MFJPY        Complete         2018-09-25T16:02:17        12/2/3                    
deploy_site        action/01CR8ZV7DHP4QK3F7KJ0AVNVG8        Complete         2018-09-25T18:28:47        12/2/3                    
deploy_site        action/01CR93DX41ZP632FP6TV8FDPMD        Complete         2018-09-25T19:31:25        17/2/1                    
deploy_site        action/01CR94DDEBAZ71P8BYQ46YW82P        Complete         2018-09-25T19:48:37        20/0/0                    
deploy_site        action/01CR97292HDNMDYDWN74PZ95HD        Complete         2018-09-25T20:34:59        20/0/0                    
deploy_site        action/01CR9837KPBRT8GSH8BCZJXS8P        Complete         2018-09-25T20:52:58        20/0/0                    
deploy_site        action/01CR98WBJT90G0BZSN32618MGX        Complete         2018-09-25T21:06:41        20/0/0                    
deploy_site        action/01CR99FRAXV3D8AEN9TESQ5FN2        Complete         2018-09-25T21:17:17        20/0/0                    
deploy_site        action/01CRBKWFJX5EFW8CJM4RR0R874        Complete         2018-09-26T18:57:29        20/0/0                    
deploy_site        action/01CRBMBNS1Q6PBAW5B16F68HFY        Complete         2018-09-26T19:05:46        20/0/0                    
deploy_site        action/01CRBVDVADJY2WHEXJYC1MBSBX        Complete         2018-09-26T21:09:18        20/0/0                    
deploy_site        action/01CRE2AHB0YHWWER8ZV6ZBHDWH        Complete         2018-09-27T17:48:19        6/1/3                     
deploy_site        action/01CRE2EHF1BS2ADS3HHYB7HYXX        Complete         2018-09-27T17:50:29        20/0/0                    
deploy_site        action/01CRE3P8WSXQT4CS70DK9J46KS        Complete         2018-09-27T18:12:11        20/0/0                    
root@ubuntu:~/deploy/site# ./run_shipyard.sh describe action/01CRE3P8WSXQT4CS70DK9J46KS
Name:                  deploy_site                                 
Action:                action/01CRE3P8WSXQT4CS70DK9J46KS           
Lifecycle:             Complete                                    
Parameters:            {}                                          
Datetime:              2018-09-27 18:12:09.757460+00:00            
Dag Status:            success                                     
Context Marker:        f65adb0b-39ca-441e-92e2-11fa5b1eb0ab        
User:                  shipyard                                    

Steps                                                                 Index        State          
step/01CRE3P8WSXQT4CS70DK9J46KS/action_xcom                           1            success        
step/01CRE3P8WSXQT4CS70DK9J46KS/dag_concurrency_check                 2            success        
step/01CRE3P8WSXQT4CS70DK9J46KS/preflight                             3            success        
step/01CRE3P8WSXQT4CS70DK9J46KS/deployment_configuration              4            success        
step/01CRE3P8WSXQT4CS70DK9J46KS/get_rendered_doc                      5            success        
step/01CRE3P8WSXQT4CS70DK9J46KS/validate_site_design                  6            success        
step/01CRE3P8WSXQT4CS70DK9J46KS/drydock_build                         7            success        
step/01CRE3P8WSXQT4CS70DK9J46KS/verify_site                           8            success        
step/01CRE3P8WSXQT4CS70DK9J46KS/armada_get_status                     9            success        
step/01CRE3P8WSXQT4CS70DK9J46KS/prepare_site                          10           success        
step/01CRE3P8WSXQT4CS70DK9J46KS/armada_build                          11           success        
step/01CRE3P8WSXQT4CS70DK9J46KS/armada_post_apply                     12           success        
step/01CRE3P8WSXQT4CS70DK9J46KS/ucp_preflight_check                   13           success        
step/01CRE3P8WSXQT4CS70DK9J46KS/promenade_validate_site_design        14           success        
step/01CRE3P8WSXQT4CS70DK9J46KS/drydock_validate_site_design          15           success        
step/01CRE3P8WSXQT4CS70DK9J46KS/armada_validate_site_design           16           success        
step/01CRE3P8WSXQT4CS70DK9J46KS/deckhand_validate_site_design         17           success        
step/01CRE3P8WSXQT4CS70DK9J46KS/prepare_and_deploy_nodes              18           success        
step/01CRE3P8WSXQT4CS70DK9J46KS/armada_get_releases                   19           success        
step/01CRE3P8WSXQT4CS70DK9J46KS/create_action_tag                     20           success        

Commands        User            Datetime                                
invoke          shipyard        2018-09-27 18:12:11.889329+00:00        
`

Reply:
{
	"dag_execution_date: '2018-08-31T22:41:52',
	u 'name': u 'update_site',
	u 'parameters': {},
	u 'action_lifecycle': u 'Complete',
	u 'datetime': u '2018-08-31 22:41:42.593915+00:00',
	u 'steps': [{
		u 'url': u '/actions/01CP92CDP1NPF8FC5AED1XK0QM/steps/action_xcom',
		u 'index': 1,
		u 'state': u 'success',
		u 'id': u 'action_xcom'
	}, 

List action is an array which will return the steps information as shown above. This was tested and it seems to be working fine for me. 

Here's the output for that test.

root@docvm:~/pk5294/airship-tempest-plugin# python -m testtools.run airship_tempest_plugin.tests.api.shipyard.test_actions.ActionsTest.test_get_action_step
Tests running...

Ran 1 test in 50.313s
OK